packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.tsx

```diff
@@ -156,6 +156,9 @@ export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
             focus();
           } else {
             nextFocusable.focus();
+
+            // Compatibility for preact where onFocus does not bubble
+            internalState.hasFocus = true;
```

Comment:
Why is this set here instead of `onRootFocusCapture`?

Reply:
also, for consistency, I think `internalState.hasFocus` should be set/unset in `onFocusCapture`/`onBlurCapture`

Reply:
I tried setting it in onRootFocusCapture, but my understanding and observed behaviour (based on [this](https://webcache.googleusercontent.com/search?q=cache:UO3E07JSSIYJ:https://harrymoreno.com/2015/12/27/event-delegation-in-javascript.html+&cd=3&hl=en&ct=clnk&gl=us)) is that the "Capture" phase goes from root to target, meaning that `onRootFocusCapture` gets called before `onBumperFocus`, which causes issues as `internalState.hasFocus` needs to be set *after* `onBumperFocus` has executed his logic. I agree that the logic as it stands is confusing, so I've made a change to remove the handler on the bumper and instead check for bumper focus directly in onRootFocusCapture. Let me know what you think!